build/scripts/DevEnv/NewDevEnv.ps1

```diff
@@ -0,0 +1,189 @@
+
+        foreach($testAppFolder in $testAppFolders) {
+            $testAppInfo = [AppProjectInfo]::Get($testAppFolder, 'test')
+            $appInfos += $testAppInfo
+        }
+    }
+
+    $script:apps = $appInfos
+    return $script:apps
+}
+
+function ResolveFolder {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $folder
+    )
+
+    if([System.IO.Path]::IsPathRooted($folder)) {
+        return $folder
+    }
+
+    return Join-Path $script:baseFolder $folder
+}
+
+function BuildApp {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $appProjectFolder,
+
+        [Parameter(Mandatory = $true)]
+        [string] $packageCacheFolder
+    )
+
+    $appProjectFolder = ResolveFolder -folder $appProjectFolder
+
+    $appInfo = [AppProjectInfo]::Get($appProjectFolder)
+    $appFile = $appInfo.GetAppFileName()
+
+    if(Test-Path (Join-Path $packageCacheFolder $appFile)) {
+        Write-Host "App $appFile already exists in $packageCacheFolder. Skipping..."
+        return (Join-Path $packageCacheFolder $appFile -Resolve)
+    }
+
+    $allAppInfos = GetAllApps
+
+    # Build dependencies
+    foreach($dependency in $appInfo.AppJson.dependencies) {
+        $dependencyAppInfo = $allAppInfos | Where-Object { $_.Id -eq $dependency.id }
+        $dependencyAppFile = BuildApp -appProjectFolder $dependencyAppInfo.AppProjectFolder -packageCacheFolder $packageCacheFolder
+        PublishApp -appFile $dependencyAppFile
+    }
+
+    $compilerFolder = CreateCompilerFolder -packageCacheFolder $packageCacheFolder
+
+    try {
+        $appFile = Compile-AppWithBcCompilerFolder -compilerFolder $compilerFolder -appProjectFolder "$($appInfo.AppProjectFolder)" -appOutputFolder $packageCacheFolder -CopyAppToSymbolsFolder -appSymbolsFolder $packageCacheFolder
+    }
+    finally {
+        Remove-Item -Path $compilerFolder -Recurse -Force | Out-Null
+    }
+
+    return $appFile
+}
+
+function PublishApp {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $appFile
+    )
+
+    Publish-BcContainerApp -containerName $script:containerName -appFile $appFile -syncMode ForceSync -sync -credential $script:credential -skipVerification -install -ignoreIfAppExists
+}
+
+$errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+Import-Module "$PSScriptRoot\..\EnlistmentHelperFunctions.psm1"
+
+$script:apps = @()
+[pscredential] $script:credential = New-Object System.Management.Automation.PSCredential ($userName, $(ConvertTo-SecureString $password -AsPlainText -Force))
+$script:baseFolder = Get-BaseFolder
+$script:packageCacheFolder = ResolveFolder -folder $packageCacheFolder
+$script:containerName = $containerName
+
+Write-Host "Loading BCContainerHelper module" -ForegroundColor Yellow
+InstallBCContainerHelper
+
+Write-Host "Creating container $script:containerName" -ForegroundColor Yellow
+CreateBCContainer
```

Comment:
Do you need the container when building the app? Setting up a container is... slow. So if there's a compilation issue with my app, it'd be nice to know about that before spinning up a container

Reply:
That order exists to potentially support compiling the apps in the container (depending on whether `useCompilerFolder` setting is used). Creating a container is needed for the dev env anyway. If an app fails to compile, the container be re-created on a subsequential run. 

If we explicitly decide to compiler the apps outside the container, then maybe creating the container and compiling the apps can be run in parallel. What do you think? 🤔